qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -1405,7 +1405,11 @@ def _decode_job(self, raw_data: Dict) -> RuntimeJob:
                 instance = to_instance_format(hub, group, project)
         # Try to find the right backend
         try:
-            backend = self.backend(raw_data["backend"], instance=instance)
+            if "backend" in raw_data:
+                backend = self.backend(raw_data["backend"], instance=instance)
+            else:
+                if self.channel == "ibm_cloud":
```

Comment:
What happens if this is false? backend will not be defined and you'll get a weirder error down below on L1430. It _shouldn't_ happen for quantum premium jobs, but it does appear to be a logic bug here.